src/main/printarray.c

```diff
@@ -348,35 +349,63 @@ void printMatrix(SEXP x, int offset, SEXP dim, int quote, int right,
     if(c > 0 && R_print.max / c < r) /* avoid integer overflow */
 	/* using floor(), not ceil(), since 'c' could be huge: */
 	r_pr = R_print.max / c;
+    int c_pr = c;
+    if (c > R_print.max) {
+        c_pr = R_print.max;
+	r_pr = 1;
```

Comment:
I think this should remain set to 0 if `r` is 0?

Reply:
Thanks for spotting the mistake. 
so adding a check for `r > 0` before setting `r_pr` to `1`  should be a good fix.